src/node/interfaces.cpp

```diff
@@ -698,14 +699,15 @@ class ChainImpl : public Chain
         limit_ancestor_count = limits.ancestor_count;
         limit_descendant_count = limits.descendant_count;
     }
-    bool checkChainLimits(const CTransactionRef& tx) override
+    util::Result<void> checkChainLimits(const CTransactionRef& tx) override
     {
-        if (!m_node.mempool) return true;
+        if (!m_node.mempool) return {};
         LockPoints lp;
         CTxMemPoolEntry entry(tx, 0, 0, 0, 0, false, 0, lp);
         LOCK(m_node.mempool->cs);
         std::string err_string;
-        return m_node.mempool->CheckPackageLimits({tx}, entry.GetTxSize(), err_string);
+        if (m_node.mempool->CheckPackageLimits({tx}, entry.GetTxSize(), err_string)) return {};
```

Comment:
Since we're now changing interfaces, I think `CheckPackageLimits` could return a `Result` type too.

Reply:
Taken in https://github.com/bitcoin/bitcoin/pull/28863/commits/e572a530d224214121371c9e42a013ef3b26ee0e
Thanks